pkg/artifactory/resource/webhook/resource_artifactory_webhook.go

```diff
@@ -357,11 +390,37 @@ func ResourceArtifactoryWebhook(webhookType string) *schema.Resource {
 			StateContext: schema.ImportStatePassthroughContext,
 		},
 
-		Schema:        domainSchemaLookup[webhookType],
+		Schema:         domainSchemaLookup(currentSchemaVersion)[webhookType],
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Type:    resourceSchemaV1().CoreConfigSchema().ImpliedType(),
+				Upgrade: ResourceStateUpgradeV1,
+				Version: 1,
+			},
+		},
+
 		CustomizeDiff: customdiff.All(
 			eventTypesDiff,
 			criteriaDiff,
 		),
 		Description:   "Provides an Artifactory webhook resource",
 	}
 }
+
+var ResourceStateUpgradeV1 = func(_ context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
```

Comment:
since these are not in a closure scope (and, I assume that's what you want) then don't use `var` - declare a normal function